pkg/subctl/operator/common/olmsubscription/utils.go

```diff
@@ -0,0 +1,59 @@
+/*
+© 2020 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package olmsubscription
+
+import (
+	"fmt"
+	"time"
+
+	//olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	olmclientv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned/typed/operators/v1alpha1"
+	//v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func WaitForReady(clientSet olmclientv1alpha1.OperatorsV1alpha1Interface, namespace string, subscription string, interval, timeout time.Duration) error {
+
+	subscriptions := clientSet.Subscriptions(namespace)
+
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		//sub, err := subscriptions.Get(subscription, metav1.GetOptions{})
+		_, err := subscriptions.Get(subscription, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return false, fmt.Errorf("error waiting for subscription: %s", err)
+		}
+		time.Sleep(120 * time.Second)
+		return true, nil
+
+		// TODO: I see "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1".SubscriptionStatus.Conditions here:
+		// https://github.com/operator-framework/operator-lifecycle-manager/blob/release-4.4/pkg/api/apis/operators/v1alpha1/subscription_types.go#L204
+		// But not on the master branch, and it fails verify:
+		// type "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1".SubscriptionStatus has no field or method Conditions
+		//for _, cond := range sub.Status.Conditions {
+		//	 TODO: I see "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1".SubscriptionCatalogSourcesUnhealthy here:
+		//	 https://github.com/operator-framework/operator-lifecycle-manager/blob/release-4.4/pkg/api/apis/operators/v1alpha1/subscription_types.go#L95
+		//	 But not on the master branch, and it fails verify:
+		//	 SubscriptionCatalogSourcesUnhealthy not declared by package v1alpha1
+		//	if cond.Type == olmv1alpha1.SubscriptionCatalogSourcesUnhealthy && cond.Status == v1.ConditionFalse {
+		//		return true, nil
+		//	}
```

Comment:
@skitt (or others) any hints about what I'm doing wrong here (as desrbied in the inline TODOs)?

Reply:
The APIs were moved from `operator-framework/operator-lifecycle-manager` to `operator-framework/api`, and are present directly in `pkg/operators` there, unversioned (at least for `subscription_types`); that’s what the OLM clientset now uses.

Reply:
Thanks @skitt, that helped! As I updated on the Automation WG call this week, I ended up realizing I needed to stick to the old 1.14.1 code due to the pinning going on elsewhere. The latest version is past this part and installing the Operator using OLM locally for me (although I'm seeing what seem to be timeouts here in CI).